File: pkg/types/app-instance.go
package types

import "time"

type AppInstance struct {
	Id        string    `json:"id"`
	AppId     string    `json:"appId"`
	Name      string    `json:"name"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

File: packages/types/src/app-instance.ts
import { z } from "zod";

export const appInstanceSchema = z.object({
    id: z.string(),
    name: z.string(),
    path: z.string(),
    description: z.string(),
    createdAt: z.coerce.date().optional(),
    updatedAt: z.coerce.date().optional(),
});

export type AppInstance = z.infer<typeof appInstanceSchema>;
